Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -80,12 +80,12 @@ final class ConnectionPool {
             if let existing = self.providers[key], existing.enqueue() {
                 return existing
             } else {
-                // Connection provider will be created with `pending = 1`
                 let provider = HTTP1ConnectionProvider(key: key,
                                                        eventLoop: taskEventLoop,
                                                        configuration: self.configuration,
                                                        pool: self,
                                                        backgroundActivityLogger: self.backgroundActivityLogger)
+                _ = provider.enqueue()
```

Comment:
yippie! Could we assert here that the return value is the expected one?

Reply:
great idea, done!